watertap/unit_models/translators/translator_asm2d_adm1.py

```diff
@@ -622,12 +629,13 @@ def XN_org(blk, t):
 
             @self.Expression(self.flowsheet().time, doc="Carbohydrates mapping")
             def Xch_mapping(blk, t):
-                return Expr_if(
-                    blk.XN_org[t] >= blk.properties_in[t].conc_mass_comp["X_S"],
-                    eps * pyunits.kg / pyunits.m**3,
+                return smooth_max(
+                    0 * pyunits.kg / pyunits.m**3,
                     (blk.properties_in[t].conc_mass_comp["X_S"] - blk.XN_org[t]) * 0.4,
+                    blk.eps,
                 )
 
+            # TODO: Can this be replaced with smooth_max or smooth_min?
             @self.Expression(self.flowsheet().time, doc="Protein mapping")
             def Xpr_mapping(blk, t):
```

Comment:
I'm not sure if this (and line 1039) can be expressed as a `smooth_max` or `smooth_min`